### ISSUE TITLE
bgb/libmobile: Fix connection timeout mechanisms

### ISSUE THREAD

**Issue by mid-kid** [REONTeam/libmobile#9]
This is a bit of a problem we're faced with when trying to synchronize two device's access to the internet, especially in the context of emulators, where the time may pass much faster than in the real world.

The problem is this: In general, calls to `connect()` on most operating systems are blocking. This means that the program halts until the connection has been established, and then resumes. This allows programs running in the background to get their things done while the network is stuck being slow/unresponsive. This is usually fine and dandy.

Enter the gameboy, an impatient little piece of (emulated) hardware that doesn't really care about these things. It'll only wait a certain amount of time on a response from the adapter before concluding "this is stuck". This interval may change depending on the speed of the emulator.

Basically, we want the adapter to stop connecting after a certain (emulated) time has passed. There's mechanisms to cause `connect()` to not block, but that will require us to busy-loop this call, as well as check if the timer has ran out. That is not pretty at all. It'd be better if we could call something like `poll()` or `select()` so the **operating system** can wait for both of these signals and schedule programs as necessary.

Additionally, if we can tie this into a mechanism where the adapter will pause the emulator to avoid the game timing out for operations that can't be done any faster, that'd be super neato. (we're talking about corner cases of insane-speedup-related corner cases here so it's not super important).
